apps/explorer/src/ui/Search.tsx

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import { Combobox } from '@headlessui/react';
+import { Search16 } from '@mysten/icons';
+
+import { LoadingSpinner } from './LoadingSpinner';
+import { Text } from './Text';
+
+type SearchResult = {
+    id: number;
+    label: string;
+};
+
+export interface SearchProps {
+    onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+    onSelectResult: (result: SearchResult) => void;
+    placeholder?: string;
+    isLoading: boolean;
+    name?: string;
+    options?: Record<string, SearchResult[]>;
+    value?: SearchResult;
+    inputValue: string;
+}
+
+export interface SearchResultProps {
+    key: string;
+    value: any;
+    children: React.ReactNode;
+}
+
+function SearchItem({ value, children }: SearchResultProps) {
+    return (
+        <Combobox.Option
+            className="cursor-pointer rounded-md bg-opacity-10 py-1.5 pl-2 ui-active:bg-sui ui-active:bg-opacity-10 ui-active:shadow-sm"
+            value={value}
+            key={value.id}
+        >
+            <Text variant="body/medium" mono color="steel-darker">
+                {children}
+            </Text>
+        </Combobox.Option>
+    );
+}
+
+export function Search({
+    onChange,
+    onSelectResult,
+    placeholder,
+    name = 'search',
+    options = {},
+    isLoading = false,
+    inputValue,
+    value,
+}: SearchProps) {
+    const hasOptions = Object.entries(options).some(
+        ([k, v]) => !!v && Object.keys(v).length
+    );
+    return (
+        <Combobox
+            value={value}
+            onChange={onSelectResult}
+            name={name}
+            data-testid={`${name}-root`}
+            as="div"
+            className="relative flex w-full flex-col"
+        >
+            <Combobox.Input
+                data-testid={`${name}-input`}
```

Comment:
Can you just select by role and label instead?

Reply:
yes